Update developer-guide.md (#13390)
dotnet/winforms: docs/developer-guide.md

## Machine Setup

Windows Forms development requires the following workloads and components be selected when installing Visual Studio 2022 (17.0.0):

* Required Workloads:
  * .NET Desktop Development
  * Desktop development with C++
* [Required Individual Components][required-individual-components]:
  * Windows 10 SDK

>   * C++ CMake tools for Windows

Reviewer: Do we not need CLI support?

Author: I don't have it installed

>   * C++/CLI support, this brings in CMake required to build tests

Reviewer: I'd explicitly list "C++ CMake tools for Windows", because that is what is technically needed.

* Required Windows Features
  * 'Media Features' including 'Windows Media Player Legacy (App)' (Windows 11) or 'Windows Media Player' (Windows 10) 
* Useful Visual Studio extensions:  
  * [Editor guidelines](https://marketplace.dev.azure.com/items?itemName=PaulHarrington.EditorGuidelines), see https://github.com/dotnet/winforms/pull/4836 for more information
  * [VS Color Output](https://marketplace.dev.azure.com/items?itemName=MikeWard-AnnArbor.VSColorOutput64)
  * [Productivity Power Tools](https://marketplace.dev.azure.com/items?itemName=VisualStudioPlatformTeam.ProductivityPowerPack2022)
 

> Optional: To get the latest C# features in the Visual Studio or to experiment with the latest runtime in the standalone IDE, install the [latest SDK build](https://github.com/dotnet/dotnet/blob/main/docs/builds-table.md).

Reviewer: ```suggestion
Optional: To get the latest C# features in Visual Studio or to experiment with the latest runtime when not using the `start-vs.cmd`, install the [latest SDK build](https://github.com/dotnet/dotnet/blob/main/docs/builds-table.md).
```
